### ISSUE TITLE
Using decoder with lm directly during inference

### ISSUE THREAD

**Issue by pdcoded** [mozilla/DeepSpeech#828]
@reuben I have written an inference script which takes the output pb file (output node being logits instead of decoded ) and then feed it to a ctc_beam_search decoder followed by spell correction for new test wav files.

I am planning to use the decoder_with_lm with pre-compiled .so files directly during inference as stated in ctc_decode_score_with_kenlm branch. 
1) Will this approach work ?
2) could not find the libctc_decoder_with_kenlm.so in the native_client folder to run this.

**Comment by elpimous** [comment]
hi.
did you do it : bazel build //native_client:ctc_decoder_with_kenlm
 libctc_decoder_with_kenlm.so should be built inside tensorflow/nazel_built/native_client/

**Comment by pdcoded** [comment]
I have generated the .so and I can use it directly in my inference now. I can share the inference gist file if anyone wants to use.

**Comment by pdcoded** [comment]
This issue can be closed now

**Comment by elpimous** [comment]
pdcoded, could you share your inference app, containing the ctc decoder ?
Thanks

**Comment by pdcoded** [comment]
Results using decoder_with_lm only in inference is worse than normal ctc decoded output with external spell correction. I had tried to optimise the hyperparameters using a grid search but the WRR decreased by 6-7%. 

Some errors include 3-4 words clubbed and missing out on initial characters.

@elpimous I will surely share my inference code.

@reuben what could be the reason? I used the .so and compiled the trie files according to instructions.

**Comment by pdcoded** [comment]
@elpimous According to comments over issue #832, native client precompiled .so file is still not leveraging the decoder with lm?

**Comment by reuben** [comment]
Currently only DeepSpeech.py uses the decoder with LM. I'm working on adding support to the native clients.

**Comment by pdcoded** [comment]
the libctc_decoder_with_kenlm.so should be agnostic to the architecture right ? I am using a variation to ds2  Can't I use the .so during inference once it is native supported ?

I am trying to interpret what the results currently are .

**Comment by reuben** [comment]
It should be, but I've no idea if it actually works, I haven't tested that setup.

**Comment by pdcoded** [comment]
Thanks @reuben. Waiting for the build. Theoretically, It should I feel. Coz I am training to get logits and feeding the logits to the decoder in inference. Coz I was able to do the same for decoder without lm.

**Comment by reuben** [comment]
This should be fixed now that #856 has landed.
